### DISCUSSION TITLE
qDate YYYY/MM without DD ( monthpicker )

### DISCUSSION THREAD

**Discussion by NicoP-S** [quasarframework/quasar#13101] [Ideas / Proposals] [answered]
Hi I would like to ask for an additional prop in the qdate component.

Sometimes i need to select a year and a month without a day of month. 
And i want to have a nice calendar like qdate has( no qselect... ) 

Some years ago there was an request for this prop...  but it was closed because it was only asked for months without years #939
Also there are requests in the forum and the discord chat.

Some solved it hacky with emit-immediately and qPopupProxy... But i need the calendar not an input with an popup.

I also did a hacky way with emit-immediately which then sets the view to months... but there is always this transition effect.
https://codesandbox.io/s/qdate-as-monthpicker-mil0e7?file=/src/pages/Index.vue 

Would it be possible to skip the day of month selection if there is a mask without a day or an additional prop to disable it?

Vuetify Example:
https://vuetifyjs.com/en/components/date-pickers-month/ 

References
https://forum.quasar-framework.org/topic/4603/how-to-select-month-and-year-only 
https://forum.quasar-framework.org/topic/7199/qdate-lock-to-select-only-yyyy-mm

**Comment by pakond** [comment]
it would be a very good improvement for the q-date of quasar I want it!

**Comment by jeangomes** [comment]
It would be awesome to have that.

**Comment by metalsadman** [comment]
try to play with this https://codepen.io/metalsadman/pen/xxYBqMo

**Reply by huiyu580** [reply]
@metalsadman 
Thank you for providing the solution. 
I'm curious about why setting the key to Date.now() when using update:model-value can  prevent switching the view to 'Calendar' in your first example.

**Reply by metalsadman** [reply]
it's to re-render the component back to its month view, anyway you can do it now with a `templateRef.value.setView('Calendar'/'Months'/'Years')`.

**Reply by huiyu580** [reply]
Thanks for your answer

**Comment by christeena-mary** [comment]
I also need this prop , it will be great improvement for q-date , since there is many date-pickers provide the same feature

**Reply by metalsadman** [reply, marked as answer]
@christeena-mary updated my codepen above https://codepen.io/metalsadman/pen/xxYBqMo

**Reply by christeena-mary** [reply]
Thank you @metalsadman

**Comment by BenJackGill** [comment]
Would also love a prop for this. I know the codepen example exists but it's not a simple solution like a prop.

**Comment by wilker818** [comment]
if you want to lock the month and year

`navigation-max-year-month`
or
`navigation-min-year-month`

**Comment by UkonnRa** [comment]
How about ranges on months / years view? Could we make the range selection to be separated from the date view so that we can have something like:

* single date picker
* single month picker
* single year picker
* range date picker
* range month picker
* range year picker
